Scripts/RemediationScripts/Remediate-CdnFrontDoorMinTLSVersion.ps1

```diff
@@ -0,0 +1,1091 @@
+                Write-Host $([Constants]::SingleDashLine)
+                if($userInput -ne "Y")
+                {
+                    Write-Host "Minimum required TLS Version will not be set for any Custom Domain. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+                    Write-Host $([Constants]::SingleDashLine)
+                    return
+                }
+                Write-Host "User has provided consent to set minimum required TLS Version on Custom Domains for all Front Doors." -ForegroundColor $([Constants]::MessageType.Update)
+                Write-Host $([Constants]::SingleDashLine)
+            }
+            else
+            {
+                Write-Host "'Force' flag is provided. Minimum required TLS Version will be set on the for all Front Doors without any further prompts." -ForegroundColor $([Constants]::MessageType.Warning)
+                Write-Host $([Constants]::SingleDashLine)
+            }
+        }
+
+        Write-Host $([Constants]::SingleDashLine)
+        Write-Host "`[Step 4 of 4] Set minimum required TLS Version for FrontDoors"
+        Write-Host $([Constants]::SingleDashLine)
+        # To hold results from the remediation.
+        $customDomainsRemediated = @()
+        $frontDoorsEPSkipped = @()
+        
+        $NonCompliantFrontDoorEndpoints | ForEach-Object {
+            $frontdoorEndpoint = $_
+            $resourceGroupName = $_.ResourceGroupName
+            $resourceName = $_.ResourceName
+            $DomainName = $_.DomainName
+            $CertificateType =$_.CertificateType
+            $PrevMinimumTlsVersion = $_.MinimumTlsVersion
+            $MinimumTlsVersion = $_.MinimumTlsVersion
+            $isMinTlsVersionSetOnCustomDomain = $_.isMinTLSVersionSetOnCustomDomain
+            $isMinTLSVersionSetOnCustomDomainPostRemediation = $false
+            
+            $frontdoorEndpoint | Add-Member -NotePropertyName isMinTLSVersionSetOnCustomDomainPostRemediation -NotePropertyValue $isMinTLSVersionSetOnCustomDomainPostRemediation
+            $frontdoorEndpoint | Add-Member -NotePropertyName PrevMinimumTlsVersion -NotePropertyValue $PrevMinimumTlsVersion
+
+            Write-Host "Setting minimum required TLS Version for custom domains of Front Door :  Resource Group Name: [$($resourceGroupName)], Resource Name: [$($resourceName)]..." -ForegroundColor $([Constants]::MessageType.Info)
+            Write-Host $([Constants]::SingleDashLine)
+            
+            if (-not [System.Convert]::ToBoolean($isMinTlsVersionSetOnCustomDomain))
+            {
+                try
+                {
+                    Write-Host "Setting minimum required TLS version on custom domain : [$DomainName]" -ForegroundColor $([Constants]::MessageType.Info)
+                    Write-Host $([Constants]::SingleDashLine)
+                    
+                    $secrets =  Get-AzFrontDoorCdnSecret -ResourceGroupName $resourceGroupName -ProfileName $resourceName
```

Comment:
This code is repeated twice in the BRS. Can we encapsulate into a private method and use it?

Reply:
Will take this as a part of another PR.